share/eds/byzantine/bad_encoding.go

```diff
@@ -155,9 +155,9 @@ func (p *BadEncodingProof) Validate(header *header.ExtendedHeader) error {
 	}
 	rebuiltShares = append(rebuiltShares, rebuiltExtendedShares...)
 
-	tree := wrapper.NewErasuredNamespacedMerkleTree(uint64(len(shares) / 2))
-	for i, share := range rebuiltShares {
-		tree.Push(share, rsmt2d.SquareIndex{Axis: uint(p.Index), Cell: uint(i)})
+	tree := wrapper.NewErasuredNamespacedMerkleTree(uint64(len(shares)/2), uint(p.Index))
```

Comment:
why p.Index instead of p.Axis ?

Reply:
good question, p.Axis only determines if its a row or column. When creating wrapped NMTs, we need to know which row or column in order to determine the correct namespace, so we use the index.

Reply:
Ah okay, I was confused bc the param is called `axisIndex` for `NewErasuredNamespacedMerkleTree`